test/integration/submission/submission-schema-migration.spec.ts

```diff
@@ -229,85 +244,293 @@ describe('schema migration api', () => {
+      await migrateSyncTo('9.0').then(async (res: any) => {
+        res.should.have.status(200);
+        const schema = (await findInDb(dburl, 'dataschemas', {})) as SchemasDictionary[];
+        // migration will fail
+        schema[0].version.should.eq(startingSchemaVersion);
+      });
+
+      const res = await getAllMigrationDocs();
+      const [migration] = res.body;
+      migration.toVersion.should.equal('9.0');
+
+      migration.stage.should.equal('FAILED');
+      migration.should.have.property('newSchemaErrors');
+      migration.newSchemaErrors.should.deep.eq({
+        [ClinicalEntitySchemaNames.DONOR]: {
+          missingFields: [DonorFieldsEnum.cause_of_death],
+          invalidFieldCodeLists: [],
+        },
+      });
+    });
+    it('should update the schema after regex and script changes invalidate donor2', async () => {
+      await migrateSyncTo('10.0').then(async (res: any) => {
+        res.should.have.status(200);
+        const schema = (await findInDb(dburl, 'dataschemas', {})) as SchemasDictionary[];
+        schema[0].version.should.eq('10.0');
+      });
+
+      const res = await getAllMigrationDocs();
+
+      const [migration] = res.body;
+      migration.toVersion.should.equal('10.0');
+
+      migration.stage.should.not.equal(migratationFailStage);
+      migration.should.not.have.property('newSchemaErrors');
+      const errorObj = migration.invalidDonorsErrors[0].errors[0];
+      errorObj.should.have.property(ClinicalEntitySchemaNames.PRIMARY_DIAGNOSIS);
+      chai
+        .expect(errorObj.primary_diagnosis[0])
+        .to.have.property('errorType', SchemaValidationErrorTypes.INVALID_BY_REGEX);
+      chai
+        .expect(errorObj.primary_diagnosis[0])
+        .to.have.property('fieldName', PrimaryDiagnosisFieldsEnum.cancer_type_code);
+
+      chai
+        .expect(errorObj.primary_diagnosis[1])
+        .to.have.property('errorType', SchemaValidationErrorTypes.INVALID_BY_SCRIPT);
+      chai
+        .expect(errorObj.primary_diagnosis[1])
+        .to.have.property('fieldName', PrimaryDiagnosisFieldsEnum.age_at_diagnosis);
+    });
+  });
+
+  describe('Prohibited changes which should be rejected', () => {
+    it('should check new schema is valid with data validation fields', async () => {
+      await migrateSyncTo('3.0');
+
+      const res = await getAllMigrationDocs();
+
+      const [migration] = res.body;
+      migration.newSchemaErrors.should.deep.eq({
+        [ClinicalEntitySchemaNames.DONOR]: {
+          missingFields: [],
+          invalidFieldCodeLists: [
+            { fieldName: DonorFieldsEnum.vital_status, missingCodeListValues: ['Deceased'] },
+          ],
+        },
+        [ClinicalEntitySchemaNames.REGISTRATION]: {
+          missingFields: [SampleRegistrationFieldsEnum.specimen_type],
+          invalidFieldCodeLists: [],
+        },
+      });
+    });
+    it.skip('should reject the new schema where the field type was changed', async () => {
```

Comment:
:eyes:, hmm this would only apply to fields that are used in validation eh. It sounds to me like the logic for checking if the schema should be rejected needs to consider the diff and not just the list of fields clinical is aware of... since this test is being skipped maybe make a ticket so it gets done (if not made already)

Reply:
I don't think there is a ticket for this, but you are correct, there probably should be (it was outlined in https://wiki.oicr.on.ca/pages/viewpage.action?spaceKey=icgcargotech&title=Submission+System+Technical+Architecture). I'll make the ticket.